src/interrogate/badge_gen.py

```diff
@@ -203,14 +206,16 @@ def should_generate_badge(output: str, color: str, result: float) -> bool:
     :return: Whether or not the badge SVG file should be generated.
     :rtype: bool
     """
-    if not os.path.exists(output):
+    output = Path(output)
+
+    if not output.exists():
         return True
 
-    if not output.endswith(".svg"):
+    if output.suffix != ".svg":
         return True
 
     try:
-        badge = minidom.parse(output)
+        badge = minidom.parse(str(output))
```

Comment:
```suggestion
        badge = minidom.parse(output)
```
minidom accepts file-like objects: https://docs.python.org/3/library/xml.dom.minidom.html#xml.dom.minidom.parse